### ISSUE TITLE
macOS 15 is not listed if the app is running outside of macOS 15

### ISSUE THREAD

**Issue by perez987** [perez987/DownloadFullInstaller#1]
macOS 15 Sequoia is in Developer Beta.

The other versions of macOS supported by the app (Big Sur, Monterey, Ventura and Sonoma), as they are not in the same phase but in official release or Public Beta, are shown in the list when the app is run on any of the systems, of course according to the Seed Program (`None`, `CustomerSeed`, `PublicSeed` or `DeveloperSeed`).

But macOS 15 is only shown in the list if the app is running from within macOS 15 with `DeveloperSeed` as Seed Program.

So far I haven't found the fix. Otherwise, the app works well.

In summary: macOS versions 11 to 14 (beta and non-beta) are always on the list but macOS 15 is only from within macOS 15.

I hope this changes when Apple releases the first macOS 15 Public Beta.

**Comment by iaihc** [comment]
14.5  run bug

-------------------------------------
Translated Report (Full Report Below)
-------------------------------------

Incident Identifier: D16BB7B2-A05F-4B96-ACCD-3C780D3E2DCA
CrashReporter Key:   037F2794-3646-26E9-4256-D48922BFA7ED
Hardware Model:      iMacPro1,1
Process:             Download Full Installer [1797]
Path:                /private/var/folders/*/Download Full Installer.app/Contents/MacOS/Download Full Installer
Identifier:          com.scriptingosx.downloadfullinstaller
Version:             1.2 (65)
Code Type:           X86-64 (Native)
Role:                Default
Parent Process:      launchd [1]
Coalition:           com.scriptingosx.downloadfullinstaller [1855]

Date/Time:           2024-07-04 08:51:41.5598 +0800
Launch Time:         2024-07-04 08:51:41.5457 +0800
OS Version:          macOS 14.5 (23F79)
Release Type:        User
Report Version:      104

Exception Type:  EXC_CRASH (SIGKILL (Code Signature Invalid))
Exception Codes: 0x0000000000000000, 0x0000000000000000
Termination Reason: CODESIGNING 1 Taskgated Invalid Signature

Triggered by Thread:  0

Thread 0 Crashed:
0                                	       0x11782c010 _dyld_start + 0
1   "                             	       0x10ffe2000 ???


Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x0000000000000000  rbx: 0x0000000000000000  rcx: 0x0000000000000000  rdx: 0x0000000000000000
  rdi: 0x0000000000000000  rsi: 0x0000000000000000  rbp: 0x0000000000000000  rsp: 0x00007ff7aff1dad8
   r8: 0x0000000000000000   r9: 0x0000000000000000  r10: 0x0000000000000000  r11: 0x0000000000000000
  r12: 0x0000000000000000  r13: 0x0000000000000000  r14: 0x0000000000000000  r15: 0x0000000000000000
  rip: 0x000000011782c010  rfl: 0x0000000000000200  cr2: 0x0000000000000000
  
Logical CPU:     0
Error Code:      0x00000000 
Trap Number:     0


Binary Images:
       0x117827000 -        0x1178b7fff  (*) <baa6f02e-dff3-3562-8c99-ea2820c91ad9> 
       0x10ffe2000 -        0x10ffe5fff " (*) <3a4cca8e-18f2-3913-b5bd-5bc398928e30> "
               0x0 - 0xffffffffffffffff ??? (*) <00000000-0000-0000-0000-000000000000> ???

Error Formulating Crash Report:
dyld_process_snapshot_get_shared_cache failed

EOF

-----------
Full Report
-----------

{"app_name":"Download Full Installer","timestamp":"2024-07-04 08:51:41.00 +0800","app_version":"1.2","slice_uuid":"3a4cca8e-18f2-3913-b5bd-5bc398928e30","build_version":"65","platform":0,"bundleID":"com.scriptingosx.downloadfullinstaller","share_with_app_devs":0,"is_first_party":0,"bug_type":"309","os_version":"macOS 14.5 (23F79)","roots_installed":0,"name":"Download Full Installer","incident_id":"D16BB7B2-A05F-4B96-ACCD-3C780D3E2DCA"}
{
  "uptime" : 4400,
  "procRole" : "Default",
  "version" : 2,
  "userID" : 501,
  "deployVersion" : 210,
  "modelCode" : "iMacPro1,1",
  "coalitionID" : 1855,
  "osVersion" : {
    "train" : "macOS 14.5",
    "build" : "23F79",
    "releaseType" : "User"
  },
  "captureTime" : "2024-07-04 08:51:41.5598 +0800",
  "codeSigningMonitor" : 0,
  "incident" : "D16BB7B2-A05F-4B96-ACCD-3C780D3E2DCA",
  "pid" : 1797,
  "cpuType" : "X86-64",
  "roots_installed" : 0,
  "bug_type" : "309",
  "procLaunch" : "2024-07-04 08:51:41.5457 +0800",
  "procStartAbsTime" : 4469163850900,
  "procExitAbsTime" : 4469177762554,
  "procName" : "Download Full Installer",
  "procPath" : "\/private\/var\/folders\/*\/Download Full Installer.app\/Contents\/MacOS\/Download Full Installer",
  "bundleInfo" : {"CFBundleShortVersionString":"1.2","CFBundleVersion":"65","CFBundleIdentifier":"com.scriptingosx.downloadfullinstaller"},
  "storeInfo" : {"deviceIdentifierForVendor":"48BAD52E-D28B-5694-882E-F84C21F12091","thirdParty":true},
  "parentProc" : "launchd",
  "parentPid" : 1,
  "coalitionName" : "com.scriptingosx.downloadfullinstaller",
  "crashReporterKey" : "037F2794-3646-26E9-4256-D48922BFA7ED",
  "codeSigningID" : "",
  "codeSigningTeamID" : "",
  "codeSigningFlags" : 16777216,
  "codeSigningValidationCategory" : 0,
  "codeSigningTrustLevel" : 4294967295,
  "sip" : "enabled",
  "exception" : {"codes":"0x0000000000000000, 0x0000000000000000","rawCodes":[0,0],"type":"EXC_CRASH","signal":"SIGKILL (Code Signature Invalid)"},
  "termination" : {"flags":66,"code":1,"namespace":"CODESIGNING","indicator":"Taskgated Invalid Signature"},
  "extMods" : {"caller":{"thread_create":0,"thread_set_state":0,"task_for_pid":0},"system":{"thread_create":0,"thread_set_state":0,"task_for_pid":0},"targeted":{"thread_create":0,"thread_set_state":0,"task_for_pid":0},"warnings":0},
  "faultingThread" : 0,
  "threads" : [{"triggered":true,"id":46295,"threadState":{"r13":{"value":0},"rax":{"value":0},"rflags":{"value":512},"cpu":{"value":0},"r14":{"value":0},"rsi":{"value":0},"r8":{"value":0},"cr2":{"value":0},"rdx":{"value":0},"r10":{"value":0},"r9":{"value":0},"r15":{"value":0},"rbx":{"value":0},"trap":{"value":0},"err":{"value":0},"r11":{"value":0},"rip":{"value":4689412112,"matchesCrashFrame":1},"rbp":{"value":0},"rsp":{"value":140701785512664},"r12":{"value":0},"rcx":{"value":0},"flavor":"x86_THREAD_STATE","rdi":{"value":0}},"frames":[{"imageOffset":20496,"symbol":"_dyld_start","symbolLocation":0,"imageIndex":0},{"imageOffset":0,"imageIndex":1}]}],
  "usedImages" : [
  {
    "source" : "P",
    "arch" : "x86_64",
    "base" : 4689391616,
    "size" : 593920,
    "uuid" : "baa6f02e-dff3-3562-8c99-ea2820c91ad9",
    "path" : "\u0004",
    "name" : "\u0004"
  },
  {
    "source" : "P",
    "arch" : "x86_64",
    "base" : 4563279872,
    "size" : 16384,
    "uuid" : "3a4cca8e-18f2-3913-b5bd-5bc398928e30",
    "path" : "\"",
    "name" : "\""
  },
  {
    "size" : 0,
    "source" : "A",
    "base" : 0,
    "uuid" : "00000000-0000-0000-0000-000000000000"
  }
],
  "vmSummary" : "ReadOnly portion of Libraries: Total=924K resident=0K(0%) swapped_out_or_unallocated=924K(100%)\nWritable regions: Total=8200K written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=8200K(100%)\n\n                                VIRTUAL   REGION \nREGION TYPE                        SIZE    COUNT (non-coalesced) \n===========                     =======  ======= \nSTACK GUARD                       56.0M        1 \nStack                             8192K        1 \nVM_ALLOCATE                          4K        1 \n__DATA                              16K        3 \n__DATA_CONST                        28K        2 \n__DATA_DIRTY                         8K        2 \n__LINKEDIT                         336K        3 \n__TEXT                             596K        2 \nmapped file                        3.9G       28 \nshared memory                        8K        2 \n===========                     =======  ======= \nTOTAL                              4.0G       45 \n",
  "legacyInfo" : {
  "threadTriggered" : {

  }
},
  "logWritingSignature" : "b0a06a1ce3951ebbead83998171bb24c7b00a78f",
  "trialInfo" : {
  "rollouts" : [
    {
      "rolloutId" : "652eff3d1bce5442b8d753c9",
      "factorPackIds" : {

      },
      "deploymentId" : 240000009
    },
    {
      "rolloutId" : "62745b8e85854550ad70b6e4",
      "factorPackIds" : {

      },
      "deploymentId" : 240000073
    }
  ],
  "experiments" : [
    {
      "treatmentId" : "45f4e2a5-551b-4bc2-a2dc-19c244dda8f8",
      "experimentId" : "6643969b3099cf28e049862f",
      "deploymentId" : 400000004
    },
    {
      "treatmentId" : "3dff9c91-a8fb-424e-a656-c8d6e6037574",
      "experimentId" : "662152ede2d11d1408c4db33",
      "deploymentId" : 400000015
    }
  ]
},
  "reportNotes" : [
  "dyld_process_snapshot_get_shared_cache failed"
]
}

Model: iMacPro1,1, BootROM 2022.100.22.0.0, 6 processors, 6-Core Intel Core i5, 2.9 GHz, 32 GB, SMC 
Graphics: AMD Radeon RX 460 (Slot 0, GPU 2), AMD Radeon RX 460, PCIe, 4 GB
Display: S24C350, 1920 x 1080 (1080p FHD - Full High Definition), Main, MirrorOff, Online
Memory Module: BANK 1/ChannelA-DIMM2, 16 GB, DDR4, 2666 MHz, A-DATA, -
Memory Module: BANK 3/ChannelB-DIMM2, 16 GB, DDR4, 2666 MHz, A-DATA, -
Bluetooth: Version (null), 0 services, 0 devices, 0 incoming serial ports
Network Service: Ethernet, Ethernet, en0
PCI Card: Intel UHD Graphics 630, Display controller, Internal@0,2,0
PCI Card: Realtek ALC S1200A, Audio device, Internal@0,31,3
PCI Card: AMD Radeon RX 460, Display Controller, Internal@0,1,0/0,0
PCI Card: Realtek RTL8125B PCI Express 2.5 Gigabit Ethernet, Ethernet controller, Internal@0,28,3/0,0
Serial ATA Device: WDC WD10EZEX-08WN4A0, 1 TB
Serial ATA Device: Samsung SSD 860 EVO 250GB, 250.06 GB
USB Device: USB30Bus
USB Device: USB2.0 Hub
USB Device: BRCM20702 Hub
USB Device: Bluetooth USB Host Controller
USB Device: AURA LED Controller
USB Device: Keyboard
USB Device: G102 LIGHTSYNC Gaming Mouse
Thunderbolt Bus:

**Comment by perez987** [comment]
@iaihc 
Please open a new issue and post there your message, it has nothing to do with the topic of this issue.
I'll try and comment, not sure if I have this problem too.

**Comment by iaihc** [comment]
![Uploading 截屏2024-07-04 15.55.55.png…]()

**Comment by iaihc** [comment]
After testing, under Sonoma 14.5, a new system version download could not be refreshed. No matter how it is set, the new version 15.0 download cannot be displayed.

**Comment by perez987** [comment]
@iaihc 
Yes, this is the issue, Sequoia beta packages only are shown if the app runs on Sequoia.
I see macOS Big Sur, Monterey, Ventura and Sonoma (beta and non beta) packages running the app on macOS Big Sur to Sonoma.But I see Sequoia beta only when running the app on the same system.
It has to do with the catalog web address, it always return

`https://swscan.apple.com/content/catalogs/others/index-15-14-13-12-10.16-10.15-10.14-10.13-10.12-10.11-10.10-10.9-mountainlion-lion-snowleopard-leopard.merged-1.sucatalog`

but Sequoia beta is listed only if catalog returns

`https://swscan.apple.com/content/catalogs/others/index-15seed-15-14-13-12-10.16-10.15-10.14-10.13-10.12-10.11-10.10-10.9-mountainlion-lion-snowleopard-leopard.merged-1.sucatalog.gz` 

I don't know hot to fix it yet. Developer @scriptongosx doesn't do anything to the repo since long time ago.

**Comment by perez987** [comment]
@iaihc 
Now that Apple has released a Public Developer Beta, the app does display macOS 15 Sequoia beta when you select developerSeed as your catalog and run the app out of macOS Sequoia (on macOS Ventura or Sonoma).

<details>
  <summary>Download Full Installer: Sequoia beta</summary>
<br>  
<img width="560" alt="DownloadFullInstaller public developer beta" src="https://github.com/user-attachments/assets/9143c3fa-14b6-44bc-8db4-9a883ce2abd6">
<br>
</details>
